src/Tools/dotnet-monitor/Commands/CollectCommandHandler.cs

```diff
@@ -143,5 +149,55 @@ private static IHostBuilder Configure(this IHostBuilder builder, StartupAuthenti
                 }
             });
         }
+
+        private static void WatchStdinForDisconnect(IServiceProvider serviceProvider, CancellationToken cancellationToken)
+        {
+            try
+            {
+                IHostApplicationLifetime lifetime = serviceProvider.GetService<IHostApplicationLifetime>();
+                using Stream inputStream = Console.OpenStandardInput();
+                byte[] buffer = new byte[100];
+
+                async Task WatchAsync()
+                {
+                    try
+                    {
+                        while (true)
+                        {
+                            if (await inputStream.ReadAsync(buffer.AsMemory(), cancellationToken) == 0)
```

Comment:
Since we are not awaiting this task, won't we dispose inputStream as soon as WatchStdinForDisconnect leaves scope? Should WatchAsync be calling Console.OpenStandardInput?

Reply:
> Since we are not awaiting this task, won't we dispose inputStream as soon as WatchStdinForDisconnect leaves scope?

No, the fact that we aren't awaiting `WatchAsync` just means that if it throws, those errors will be dropped on the floor. But otherwise, `await` has no effect. If dotnet-monitor has a prefered way of marking a task as ignored, I am happy to switch to that. I searched around, but I couldn't find anything except one or two other places where the task was assigned to the discard variable. 

> Should WatchAsync be calling Console.OpenStandardInput?

I didn't want to do that, as any exception from `WatchAsync` is dropped on the floor (since we can't await it). So I wanted to alert the caller to the fact that they specified `--exit-on-stdin-disconnect` but their stdin was corrupt, so it wouldn't work.

Reply:
Seems like I should probably at least add a comment to line 195 (`_ = WatchAsync()` line).

Reply:
I think Wiktor's first point is that since you are not awaiting the `Task` returned from `WatchAsync`, then the `WatchStdinForDisconnect` method immediately exits after calling `WatchAsync`. Since `using Stream inputStream = Console.OpenStandardInput();` is in the body of `WatchStdinForDisconnect`, it will be nearly immediately disposed. This also means the try...catch isn't really going to catch much unless it happens in the opening of `WatchStdinForDisconnect` before the first await.

Reply:
Ah. Thanks. That is what I was attempting with the catch, but certainly not the stream.